tensorflow/core/user_ops/gather_columns.cc

```diff
@@ -0,0 +1,183 @@
+#include "tensorflow/core/framework/op.h"
+//#include "tensorflow/core/framework/shape_inference.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/register_types.h"
+#include "tensorflow/core/platform/prefetch.h"
+
+using namespace tensorflow;
+using namespace std;
+
+
+REGISTER_OP("GatherColumns")
+    .Input("params: T")
+    .Input("indices: Index")
+    .Output("columns: T")
+    .Attr("T: type")
+    .Attr("Index: {int32,int64}");
+
+template <typename T>
+class GatherColumnsOp : public OpKernel {
+ public:
+  explicit GatherColumnsOp(OpKernelConstruction* ctx) : OpKernel(ctx) {}
+
+  void Compute(OpKernelContext* ctx) override {
+
+    //--Grab the input tensor - params--//
+    const Tensor& params = ctx->input(0);
+
+    //--Grab the input tensor - indices--//
+    const Tensor& indices = ctx->input(1);
+    auto ind_flat = indices.flat<int32>();
+
+    OP_REQUIRES(ctx, TensorShapeUtils::IsVectorOrHigher(params.shape()),
+                errors::InvalidArgument("params must be at least a vector"));
+    OP_REQUIRES(ctx, TensorShapeUtils::IsVector(indices.shape()),
+                errors::InvalidArgument("indices must be a vector, but it a: ", indices.dims(), "D Tensor."));
```

Comment:
Not sure what you mean, I simply meant that it seems to be missing "is" as in "but it IS a".